addons/account_payment/controllers/payment.py

```diff
@@ -32,7 +32,7 @@ def invoice_transaction(self, invoice_id, access_token, **kwargs):
 
         logged_in = not request.env.user._is_public()
         partner_sudo = request.env.user.partner_id if logged_in else invoice_sudo.partner_id
-        self._validate_transaction_kwargs(kwargs)
+        self._validate_transaction_kwargs(kwargs, {'installment_number'})
```

Comment:
Always pass kwargs as kwargs.
```suggestion
        self._validate_transaction_kwargs(kwargs, additional_allowed_keys={'installment_number'})
```